backend/models/user.js

```diff
@@ -0,0 +1,18 @@
+const mongoose = require('mongoose')
+const Schema = mongoose.Schema;
+const User = new Schema({
+        username: String,
```

Comment:
I would change this to
username: {
   type: String,
   unique: true
}

so we can later have a userProfile schema for the teachers that refences the username as a unique key

Reply:
Apologies... I realized that we are not requiring the user to pick a unique username in the signup process. So a better choice could be to use the signup email as the unique key and not include the username in the schema

Reply:
you're right with that, have set the suggested change with email, so the email will be considered as unique.